src/client/Microsoft.Identity.Client/Internal/Requests/Silent/SilentRequest.cs

```diff
@@ -61,8 +61,9 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
                        UiRequiredExceptionClassification.AcquireTokenSilentFailed);
                 }
 
-                bool isAccountSourceDeviceCodeFlow = !string.IsNullOrEmpty(AuthenticationRequestParameters.Account.AccountSource) &&
-                                               AuthenticationRequestParameters.Account.AccountSource == "device_code_flow";
+                var account = AuthenticationRequestParameters.Account as Account;
+                bool isAccountSourceDeviceCodeFlow = !string.IsNullOrEmpty(account.AccountSource) &&

```

Comment:
`AccountSource` property can be changed to an enum of type `ApiIds`

Reply:
I will leave it to a free from string for now. Also MSAL.NET only has to skip broker for device code flow. ROPC is already through broker in MSAL.NET. I in fact want to get rid of this workarounds in the future when device code flow is deprecated and not make it easy to introduce the workarounds for any other flow by mistake.